### DISCUSSION TITLE
Allowing web testing (or user-submitted examples)

### DISCUSSION THREAD

**Discussion by TomLucidor** [OpenModelDB/open-model-database#323] [General]
Currently there are some models without images (e.g. the JPEG denoiser) and that makes model selection a bit annnoying.
Some suggestion:
1. Allow people to share examples of different upscalers being used, but that would require something similar to CivitAI posts
2. (might be hard) allow direct GPU-based browser upscaling for testing multiple upscalers at once and cross-compare
3. Requiring submissions to test against standard set of blurry or low-resolution images for different usecases

**Comment by RunDevelopment** [comment]
> Allow people to share examples of different upscalers being used, but that would require something similar to CivitAI posts

This will be difficult. User submissions typically require (1) having a login system and (2) moderating user submissions. (1) requires time to implement and get right (security), and (2) is an ongoing maintenance cost.

> (might be hard) allow direct GPU-based browser upscaling for testing multiple upscalers at once and cross-compare

@joeyballentine is currently looking into that. GPU support might be difficult, doing it on the CPU is definitely possible.

> Requiring submissions to test against standard set of blurry or low-resolution images for different usecases

This is actually planned, but as a more automate process. The idea is that we define a standard set of images for each tag, and some process automatically adds the necessary upscaled images.

**Reply by TomLucidor** [reply]
I would like to locally test with some of my own images, if only there is a way to download all the upscalers for fast comparisons (or maybe browser-based tools that does not log the user)
